packages/date-picker/src/FilterDateRangePicker/hooks/useRangeDateValidation.spec.tsx

```diff
@@ -1,99 +1,74 @@
+import { useRangeDateValidation } from "./useRangeDateValidation"
 
-const Wrapper = ({
-  date,
-  inputValue,
-  ...restProps
-}: Partial<UseRangeDateValidationArgs> & {
-  date: Date | undefined
-  inputValue: string
-}): JSX.Element => {
-  const [newDate, setNewDate] = useState<Date | undefined>()
-  const { status, validationMessage, validateDate, updateValidation } =
-    useRangeDateValidation({
-      inputLabel: "Start date",
-      ...restProps,
-    })
-
-  return (
-    <div>
-      <p data-testid="new-date">{newDate ? newDate.toJSON() : "undefined"}</p>
-      <p data-testid="status">{status ?? "undefined"}</p>
-      <p data-testid="message">{validationMessage ?? "undefined"}</p>
-      <button
-        type="button"
-        onClick={(): void => {
-          const { validationResponse, newDate: responseNewDate } = validateDate(
-            {
-              date,
-              inputValue,
-            }
-          )
-          updateValidation(validationResponse)
-          setNewDate(responseNewDate)
-        }}
-      >
-        Validate
-      </button>
-    </div>
-  )
-}
-
-const validateValue = (): void => {
-  userEvent.click(screen.getByRole("button", { name: "Validate" }))
-}
-
-describe("useRangeDateValidation", () => {
-  test("valid date", async () => {
-    render(<Wrapper date={new Date("2022-05-01")} inputValue="1 May 2022" />)
-    validateValue()
-
-    await waitFor(() => {
-      expect(screen.getByTestId("new-date").textContent).toBe(
-        "2022-05-01T00:00:00.000Z"
+describe("useRangeDateValidation()", () => {
+  describe("with a valid date", () => {
+    it("returns no error status, no validation message and the same date", () => {
+      const { result } = renderHook(() =>
+        useRangeDateValidation({
+          inputLabel: "Start date",
+        })
       )
-      expect(screen.getByTestId("status").textContent).toBe("undefined")
-      expect(screen.getByTestId("message").textContent).toBe("undefined")
+      const { validateDate } = result.current
+      const { newDate } = validateDate({
+        date: new Date("2022-05-01"),
+        inputValue: "1 May 2022",
+      })
+
+      expect(result.current.status).toBeUndefined()
+      expect(result.current.validationMessage).toBeUndefined()
+      expect(newDate).toEqual(new Date("2022-05-01"))
     })
   })
 
-  test("invalid date", async () => {
-    render(<Wrapper date={new Date("potato")} inputValue="potato" />)
-    validateValue()
+  describe("with an invalid date", () => {
+    it("returns an error status, a validation message and no date", () => {
+      const { result } = renderHook(() =>
+        useRangeDateValidation({
+          inputLabel: "Start date",
+        })
+      )
+      const { validateDate, updateValidation } = result.current
+      const { validationResponse, newDate } = validateDate({
+        date: new Date("potato"),
+        inputValue: "potato",
+      })
+
+      act(() => {
```

Comment:
Should it be this instead? Or are hooks different? You'd need to `await` regardless though I think (at least, I know that `waitFor` without it would often lead to false positives).

```suggestion
      await waitFor(() => {
```

Reply:
iirc The test was working but complaining that something was doing something to the DOM and should be wrapped in `act` - so I did that to remove the warning.
I just tested it with `waitFor` - it actually makes the test fail, so we might just leave it with `act`.

Reply:
Maybe hooks need to be tested differently. All good!